src/Symfony/Bundle/SecurityBundle/Routing/LogoutRouteLoader.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Routing;
+
+use Symfony\Component\DependencyInjection\Config\ContainerParametersResource;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+class LogoutRouteLoader
+{
+    /**
+     * @param iterable<string, string> $logoutPaths   Logout paths indexed by the corresponding firewall name
+     * @param string                   $parameterName Name of the container parameter containing {@see $logoutPaths}' value
+     */
+    public function __construct(
+        private readonly iterable $logoutPaths,
+        private readonly string $parameterName,
+    ) {
+    }
+
+    public function __invoke(): RouteCollection
+    {
+        $collection = new RouteCollection();
+        $collection->addResource(new ContainerParametersResource([$this->parameterName => $this->logoutPaths]));
+
+        $routeNames = [];
+        foreach ($this->logoutPaths as $firewallName => $logoutPath) {
+            $routeName = '_logout_'.$firewallName;
+
+            if (isset($routeNames[$logoutPath])) {
+                $collection->addAlias($routeNames[$logoutPath], $routeName);
+            } else {
+                $routeNames[$logoutPath] = $routeName;
+            }
+
+            $collection->add($routeName, new Route($logoutPath));
```

Comment:
shouldn't this be in the `else` block instead of adding a route when we already add an alias ?

Reply:
I was surprised about that too, but `RouteCollection::addAlias` will remove the original route, so I have to add it anyways: https://github.com/symfony/symfony/blob/33fedcee2814947c59df2570952f8c800dfdb23e/src/Symfony/Component/Routing/RouteCollection.php#L372-L381

Reply:
Not sure to understand; is there something left to do here? :sweat_smile:

Reply:
ah no. this is even more weird.

Your current logic actually replace the **previous** route with an alias to the newly added route (and you might end up with a chain of aliases if this gets done multiple times)

Reply:
@MatTheCat you could also achieve that by defining an alias for the first occurrence of the logout path and then defining next routes as aliases of the first one. this would be much simpler than defining the last firewall as a route and previous ones as aliases.

currently, you are defining this in the end:

```
_logout_firewall1:
  alias: _logout_firewall2

_logout_firewall2:
  alias: _logout_firewall3

_logout_firewall3:
  alias: _logout_firewall4

_logout_firewall4:
  path: /logout
```

the more firewall you have with the same logout path, the longer the alias chain becomes to generate the logout URL for the first one.

Reply:
And re-reading the code, I'm not even sure that this is what happens here as I think you will actually end up with the first firewall being defining as an alias to the last route and all other firewalls as actual routes (as `$routeNames[$logoutPath]` will stay the name of the route of the first firewall, and this is the alias being redefined constantly).

My guess is that you confused the meaning of the 2 parameters of `addAlias`

Reply:
Spot-on analysis! Should be better with https://github.com/symfony/symfony/pull/50946/commits/f9dba15a020219be32fd2bbd4a223f6b40dbb7fc